src/dbus_api/api.rs

```diff
@@ -56,27 +57,31 @@ fn create_pool(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
 
     let object_path = m.path.get_name();
     let dbus_context = m.tree.get_data();
-    let result = dbus_context
-        .engine
-        .borrow_mut()
-        .create_pool(name, &blockdevs, tuple_to_option(redundancy), force);
+    let mut engine = dbus_context.engine.borrow_mut();
+    let result = engine.create_pool(name, &blockdevs, tuple_to_option(redundancy), force);
 
     let return_message = message.method_return();
 
     let msg = match result {
-        Ok((uuid, devnodes)) => {
+        Ok(pool_uuid) => {
             let pool_object_path: dbus::Path =
-                create_dbus_pool(dbus_context, object_path.clone(), uuid);
-            let paths = devnodes
-                .iter()
-                .map(|d| {
-                         d.to_str()
-                             .expect("'d' originated in the 'devs' D-Bus argument")
-                             .into()
-                     });
-            let paths = paths.map(MessageItem::Str).collect();
+                create_dbus_pool(dbus_context, object_path.clone(), pool_uuid);
+            let default_return =
+                MessageItem::Struct(vec![MessageItem::ObjectPath(default_object_path()),
+                                         MessageItem::Array(vec![], "o".into())]);
+            let pool_uuid_ref = &pool_uuid;
+            let pool = get_mut_pool!(engine; pool_uuid_ref; default_return; return_message);
+
+            let mut bd_object_paths = Vec::new();
```

Comment:
Apparently, you can substitute a map() for this for loop, like you do down below.

Reply:
ok will do.